benches/benchmark_portscan.rs

```diff
@@ -23,17 +24,18 @@ fn portscan_tcp() {
         vec![],
         false,
     );
-    // Perform the actual scan or logic here if needed
+
+    let _ = block_on(scanner.run());
 }
 
 fn portscan_udp() {
     let addrs = vec!["127.0.0.1".parse::<IpAddr>().unwrap()];
     let range = PortRange {
         start: 1,
-        end: 60_000,
+        end: 1023,
     };
     let strategy = PortStrategy::pick(&Some(range), None, ScanOrder::Serial);
-    let _scanner = Scanner::new(
+    let scanner = Scanner::new(
         &addrs,
         10,
         Duration::from_millis(100),
```

Comment:
The issue with this is that it creates the range, port strategy and parses IP addresses in the benchmark too. I will try to revive my laptop today but I moved this out of these functions to only benchmark the scanning :D

Reply:
Yep, I see what you mean.

I'll tinker with this but equally, if you manage to resurrect your laptop, I'll happily junk this PR.

Reply:
hahahah i am praying to the Mesa graphics drivers gods 🙏🏻